src/platform/mock-local-home-platform.ts

```diff
@@ -1,144 +1,91 @@
-    this.newDeviceRegisteredActions.forEach(newDeviceRegisteredAction => {
-      newDeviceRegisteredAction(localDeviceId);
-    });
-  }
-
-  public async getNextDeviceIdRegistered(): Promise<string> {
-    return new Promise(resolve => {
-      this.newDeviceRegisteredActions.push(localDeviceId => {
-        resolve(localDeviceId);
-      });
-    });
-  }
-
-  public isHomeAppReady(): boolean {
-    return this.homeAppReady;
+  public getDeviceManager(): smarthome.DeviceManager {
+    return this.deviceManager;
   }
 
-  public notifyHomeAppReady(): void {
-    this.homeAppReady = true;
+  public getLocalDeviceIdMap(): Map<string, string> {
+    return this.localDeviceIds;
   }
 
-  //  Singleton getter
-  public static getInstance(): MockLocalHomePlatform {
-    if (!MockLocalHomePlatform.instance) {
+  /**
+   * Returns the static singleton instance, creating it if needed.
+   * @param resetState  Whether or not to force a new instance.  Useful for initializing tests.
+   */
+  public static getInstance(
+    resetState: boolean = false
+  ): MockLocalHomePlatform {
+    if (!MockLocalHomePlatform.instance || resetState) {
       MockLocalHomePlatform.instance = new MockLocalHomePlatform();
     }
     return MockLocalHomePlatform.instance;
   }
 
-  public getDeviceManager(): smarthome.DeviceManager {
-    return this.deviceManager;
-  }
-
-  public getLocalDeviceIdMap(): Map<string, string> {
-    return this.localDeviceIds;
+  private onNewDeviceIdRegistered(localDeviceId: string) {
+    this.newDeviceRegisteredActions.forEach(newDeviceRegisteredAction => {
+      newDeviceRegisteredAction(localDeviceId);
+    });
   }
 
-  private setupUDP() {
-    this.udpScanConfigs.forEach(udpScanConfig => {
-      this.mockNetwork!.registerUDPListener(
-        this,
-        udpScanConfig.listenPort,
-        udpScanConfig.broadcastAddress
-      );
+  /**
+   * Asyncronously returns the next localDeviceId registered to the Local Home Platform.
+   * This localDeviceId is referred to as the verificationId in the IdentifyResponse
+   */
+  public async getNextDeviceIdRegistered(): Promise<string> {
+    return new Promise(resolve => {
+      this.newDeviceRegisteredActions.push(localDeviceId => {
+        resolve(localDeviceId);
+      });
     });
   }
 
-  private sendUDPBroadcast(scanConfig: UDPScanConfig) {
-    const packetBuffer = Buffer.from(scanConfig.discoveryPacket, 'hex');
-    this.mockNetwork!.sendUDPMessage(
-      packetBuffer,
-      scanConfig.broadcastPort,
-      scanConfig.broadcastAddress,
-      scanConfig.listenPort,
-      scanConfig.broadcastAddress
-    );
-  }
+  /**
+   * Takes a `discoveryBuffer` and passes it to the fulfillment app in an `IdentifyRequest`
+   * @param discoveryBuffer  the buffer to be included in the `IdentifyRequest` scan data
+   */
+  public async triggerIdentify(discoveryBuffer: Buffer): Promise<void> {
+    console.log('Received discovery payload:', discoveryBuffer);
```

Comment:
Do we want to keep the logging? maybe set it as `.debug`?

Reply:
Done